plugins/modules/web_infrastructure/jira.py

```diff
@@ -662,35 +410,258 @@ def main():
+
+    @cause_changes(on_success=True)
+    def operation_comment(self):
+        data = {
+            'body': self.vars.comment
+        }
+        url = self.vars.restbase + '/issue/' + self.vars.issue + '/comment'
+        self.vars.meta = self.post(url, data)
+
+    @cause_changes(on_success=True)
+    def operation_edit(self):
+        data = {
+            'fields': self.vars.fields
+        }
+        url = self.vars.restbase + '/issue/' + self.vars.issue
+        self.vars.meta = self.put(url, data)
+
+    @cause_changes(on_success=True)
+    def operation_update(self):
+        data = {
+            "update": self.vars.fields,
+        }
+        url = self.vars.restbase + '/issue/' + self.vars.issue
+        self.vars.meta = self.put(url, data)
+
+    def operation_fetch(self):
+        url = self.vars.restbase + '/issue/' + self.vars.issue
+        self.vars.meta = self.get(url)
+
+    def operation_search(self):
+        url = self.vars.restbase + '/search?jql=' + pathname2url(self.vars.jql)
+        if self.vars.fields:
+            fields = self.vars.fields.keys()
+            url = url + '&fields=' + '&fields='.join([pathname2url(f) for f in fields])
+        if self.vars.maxresults:
+            url = url + '&maxResults=' + str(self.vars.maxresults)
+
+        self.vars.meta = self.get(url)
+
+    @cause_changes(on_success=True)
+    def operation_transition(self):
+        # Find the transition id
+        turl = self.vars.restbase + '/issue/' + self.vars.issue + "/transitions"
+        tmeta = self.get(turl)
+
+        target = self.vars.status
+        tid = None
+        for t in tmeta['transitions']:
+            if t['name'] == target:
+                tid = t['id']
+                break
+        else:
+            raise ValueError("Failed find valid transition for '%s'" % target)
+
+        fields = dict(self.vars.fields)
+        if self.vars.summary is not None:
+            fields.update({'summary': self.vars.summary})
+        if self.vars.description is not None:
+            fields.update({'description': self.vars.description})
+
+        # Perform it
+        data = {'transition': {"id": tid},
+                'fields': fields}
+        if self.vars.comment is not None:
+            data.update({"update": {
+                "comment": [{
+                    "add": {"body": self.vars.comment}
+                }],
+            }})
+        url = self.vars.restbase + '/issue/' + self.vars.issue + "/transitions"
+        self.vars.meta = self.post(url, data)
+
+    @cause_changes(on_success=True)
+    def operation_link(self):
+        data = {
+            'type': {'name': self.vars.linktype},
+            'inwardIssue': {'key': self.vars.inwardissue},
+            'outwardIssue': {'key': self.vars.outwardissue},
+        }
+        url = self.vars.restbase + '/issueLink/'
+        self.vars.meta = self.post(url, data)
+
+    @cause_changes(on_success=True)
+    def attach(self):
```

Comment:
```suggestion
    def operation_attach(self):
```
Otherwise this won't work I think.